tests/util/icon/test_timing_tree.py

```diff
@@ -29,167 +28,142 @@
-            "root>exch_data>exch_data.wait",
-            "root>total>integrate_nh>nh_solve",
-            "root>total>integrate_nh",
-            "root>total>integrate_nh>physics>nwp_radiation>nwp_ecrad_utilities",
-            "root>total",
-            "root>total>integrate_nh>nh_solve>nh_solve.vimpl",
-            "root>total>integrate_nh>physics>nwp_turbulence",
-            "root>total>write_restart>write_restart_communication",
-            "root>total>write_restart>write_restart_io",
-            "root>total>integrate_nh>physics>diagnose_pres_temp",
-            "root>total>integrate_nh>physics>nwp_microphysics",
-            "root>total>integrate_nh>physics>nwp_surface",
-            "root>total>integrate_nh>physics>nwp_convection",
-            "root>total>integrate_nh>physics>nwp_radiation>preradiaton",
-            "root>total>integrate_nh>nh_solve>nh_solve.vnupd",
-            "root>total>integrate_nh>physics>nwp_radiation>nwp_radiation_downscale",
-            "root>total>integrate_nh>nesting",
-            "root>total>integrate_nh>transport>adv_vert",
-            "root>total>integrate_nh>transport",
-            "root>total>integrate_nh>physics>phys_u_v",
-            "root>total>integrate_nh>physics>satad",
-            "root>total>integrate_nh>physics>phys_acc_sync",
-            "root>total>integrate_nh>physics>radheat",
-            "root>total>integrate_nh>physics>nwp_radiation>nwp_ecrad_ecrad",
-        }
-
-        self.assertTrue(
-            names == ref_names, msg="set of intersecting nodes does not match reference"
-        )
-
-    def test_sub(self):
-        tt1 = TimingTree.from_logfile(TIMING_FILE_1, read_logfile)
-        tt2 = TimingTree.from_logfile(TIMING_FILE_2, read_logfile)
-
-        sub_nodes = tt1.root[-1].sub(tt2.root[-1])
-
-        names = {n.get_ancestry_name() for n in sub_nodes}
-
-        ref_names = {"root>total>integrate_nh>nh_solve>nh_solve.edgecomp"}
-
-        self.assertTrue(
-            names == ref_names, msg="set of differing nodes does not match reference"
-        )
-
-    def test_grow(self):
-        tt1 = TimingTree.from_logfile(TIMING_FILE_1, read_logfile)
-        tt2 = TimingTree.from_logfile(TIMING_FILE_2, read_logfile)
-
-        diff_nodes = tt1.root[-1].sub(tt2.root[-1])
-
-        tt2.grow(diff_nodes, -1)
-
-        new_node = tt2.find("nh_solve.edgecomp", -1)
-
-        self.assertEqual(
-            new_node.name,
-            "nh_solve.edgecomp",
-            msg="did not add (non-present) node nh_solve.edgecomp",
-        )
-
-    def test_add(self):
-        tt1 = TimingTree.from_logfile(TIMING_FILE_1, read_logfile)
-        tt2 = TimingTree.from_logfile(TIMING_FILE_2, read_logfile)
-        tt_added = TimingTree.from_json(JSON_ADD_REFERENCE)
-
-        tt1.add(tt2)
-
-        self.assert_trees_equal(tt1, tt_added)
-
-
-if __name__ == "__main__":
-    unittest.main()
+@pytest.fixture(name="timing_tree")
+def fixture_timing_tree():
+    return TimingTree
```

Comment:
Why do you inject `TimingTree` via a fixture into the test-functions and not simply through the import in L. 14. itself?
Does not make sense to me

Reply:
Good point. Thanks.